src/software/logger/logger.h

```diff
@@ -58,13 +58,16 @@ class LoggerSingleton
      *
      * @param runtime_dir The directory where the log files will be stored.
      */
-    static void initializeLogger(const std::string& runtime_dir)
+    static void initializeLogger(const std::string& runtime_dir,
+                                 const std::shared_ptr<ProtoLogger>& proto_logger)
     {
-        static std::shared_ptr<LoggerSingleton> s(new LoggerSingleton(runtime_dir));
+        static std::shared_ptr<LoggerSingleton> s(
+            new LoggerSingleton(runtime_dir, proto_logger));
```

Comment:
shouldn't this be `std::unique_ptr` instead of `std::shared_ptr`? 

The intention of this is to only initialize this once, am I right?

Reply:
By convention, aren't we using `std::make_shared` or `std::make_unique`?

Reply:
> shouldn't this be std::unique_ptr instead of std::shared_ptr?

You might be able to use a unique_ptr, I haven't tried it. Though we do call the logger in many places (though without taking ownership), so not fully sure if it would work.

> By convention, aren't we using std::make_shared or std::make_unique?

Yes, this has been in the codebase for while so I haven't touched. But if you want I could try updating it.